File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/model/Currency.java
package org.mifos.mobilewallet.core.domain.model;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by naman on 17/8/17.
 */

public class Currency implements Parcelable {

    public static final Creator<Currency> CREATOR = new Creator<Currency>() {
        @Override
        public Currency createFromParcel(Parcel source) {
            return new Currency(source);
        }

        @Override
        public Currency[] newArray(int size) {
            return new Currency[size];
        }
    };
    private String code;
    private String displaySymbol;
    private String displayLabel;

    public Currency() {
    }

    protected Currency(Parcel in) {
        this.code = in.readString();
        this.displaySymbol = in.readString();
        this.displayLabel = in.readString();
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getDisplaySymbol() {
        return displaySymbol;
    }

    public void setDisplaySymbol(String displaySymbol) {
        this.displaySymbol = displaySymbol;
    }

    public String getDisplayLabel() {
        return displayLabel;
    }

    public void setDisplayLabel(String displayLabel) {
        this.displayLabel = displayLabel;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.code);
        dest.writeString(this.displaySymbol);
        dest.writeString(this.displayLabel);
    }
}

File: core/model/src/main/java/com/mifos/mobilewallet/model/entity/client/Currency.kt
package com.mifos.mobilewallet.model.entity.client

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class Currency (
    @SerializedName("code")
    var code: String?=null,

    @SerializedName("name")
    var name: String?=null,

    @SerializedName("decimalPlaces")
    var decimalPlaces: Int?=null,

    @SerializedName("displaySymbol")
    var displaySymbol: String?=null,

    @SerializedName("nameCode")
    var nameCode: String?=null,

    @SerializedName("displayLabel")
    var displayLabel: String?=null,
    ) : Parcelable {
    constructor() : this(null, null, null, null, null, null)
    }